<a href="https://colab.research.google.com/github/akarshpyr/Python-codes/blob/main/Hypothesis_assignment_Q1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Hypothesis testing problem 1 (Cutlet diameter test)

The status quo assumed is that there is no difference in the sizes of cutlets.

Null hypothesis is Mean 1 = Mean 2; 
 Alternate hypothesis is Mean1 != Mean2.

In [1]:
import pandas as pd
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df = pd.read_csv('/content/Cutlets.csv')

In [3]:
df

Unit A  Unit B
0   6.8090  6.7703
1   6.4376  7.5093
2   6.9157  6.7300
3   7.3012  6.7878
4   7.4488  7.1522
5   7.3871  6.8110
6   6.8755  7.2212
7   7.0621  6.6606
8   6.6840  7.2402
9   6.8236  7.0503
10  7.3930  6.8810
11  7.5169  7.4059
12  6.9246  6.7652
13  6.9256  6.0380
14  6.5797  7.1581
15  6.8394  7.0240
16  6.5970  6.6672
17  7.2705  7.4314
18  7.2828  7.3070
19  7.3495  6.7478
20  6.9438  6.8889
21  7.1560  7.4220
22  6.5341  6.5217
23  7.2854  7.1688
24  6.9952  6.7594
25  6.8568  6.9399
26  7.2163  7.0133
27  6.6801  6.9182
28  6.9431  6.3346
29  7.0852  7.5459
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875

In [4]:
df.describe()

Unit A     Unit B
count  35.000000  35.000000
mean    7.019091   6.964297
std     0.288408   0.343401
min     6.437600   6.038000
25%     6.831500   6.753600
50%     6.943800   6.939900
75%     7.280550   7.195000
max     7.516900   7.545900

Apply a t-test as sample is small.
The degress of freedom is found manually for a 2 sample independent t-test. 
dof = 66.

In [5]:
t_critical = stats.t.ppf(q=1-0.05/2, df=67)

In [7]:
t_critical

1.9960083534755055

Find the t-statistic value and probablity.

In [8]:
t_statistic_p_value = stats.ttest_ind(df['Unit A'].values, df['Unit B'].values)

In [9]:
t_statistic_p_value

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

Since the p_value is greater than the significance level alpha (0.47>0.05) we would fail to reject null hypothesis.
The status quo assumed would be upheld.